### PR TITLE
Enable extension of the dce phase

### PR DESCRIPTION
**Pull request by davleopo**
Change interface of the dominator based conditional elimination.
Please review: @thomaswue

### CODE REVIEW

File: graal/com.oracle.graal.phases.common/src/com/oracle/graal/phases/common/DominatorConditionalEliminationPhase.java [hunk 1]
```diff
@@ -229,655 +172,720 @@ public void processBlock(Block startBlock) {
...
+                return tryProofGuardCondition(null, node, rewireGuardFunction);
+            }
 
-                ValueNode x = binaryOpLogicNode.getX();
-                ValueNode y = binaryOpLogicNode.getY();
-                for (InfoElement infoElement : getInfoElements(x)) {
-                    TriState result = binaryOpLogicNode.tryFold(infoElement.getStamp(), y.stamp());
-                    if (result.isKnown()) {
-                        return rewireGuards(infoElement.getGuard(), result.toBoolean(), rewireGuardFunction);
+            protected boolean tryProofGuardCondition(DeoptimizingGuard thisGuard, LogicNode node, GuardRewirer rewireGuardFunction) {
```
Comment (dougxc) [inline graal/com.oracle.graal.phases.common/src/com/oracle/graal/phases/common/DominatorConditionalEliminationPhase.java]: `tryProofGuardCondition` -> `tryProveGuardCondition`

**Comment by dougxc** [comment]
While this seems like a lot of changes, I think it's just a reorganization and refactoring of code to be able to customize DCE right?

**Comment by davleopo** [comment]
yes exactly, it changes the previously static methods for folding stamps  to virtual ones in the instance frame and adjusts the visibility

File: graal/com.oracle.graal.phases.common/src/com/oracle/graal/phases/common/DominatorConditionalEliminationPhase.java (hunk 1 above)
Comment (davleopo) [inline graal/com.oracle.graal.phases.common/src/com/oracle/graal/phases/common/DominatorConditionalEliminationPhase.java]: thanks, will change that

**Comment by dougxc** [comment]
Ok. Can you make the renaming I pointed out (which I know you're not responsible for) and then I'll merge this PR.

**Comment by davleopo** [comment]
yes, thank you.

**Pull request merged**